src/arch/svm/utils.ts

```diff
@@ -49,6 +49,30 @@ export function toAddress(address: SdkAddress): Address<string> {
   return address.toBase58() as Address<string>;
 }
 
+/**
+ * Resolve the latest finalized slot, and then work backwards to find the nearest slot containing a block.
+ * In most cases the first-resolved slot should also have a block. Avoid making arbitrary decisions about
+ * how many slots to rotate through.
+ */
+export async function getLatestFinalizedSlotWithBlock(provider: SVMProvider, maxLookback = 1000): Promise<number> {
+  let slot = await provider.getSlot({ commitment: "finalized" }).send();
+  const endSlot = slot;
+  const opts = { maxSupportedTransactionVersion: 0, transactionDetails: "none", rewards: false } as const;
+
+  do {
+    const block = (await provider.getBlock(slot, opts).send());
+    if (isDefined(block)) {
```

Comment:
It seems like the block can be defined but the timestamp is undefined (because of the return value [here](https://solana.com/ja/docs/rpc/http/getblock)) which is not what we want. Do you think this is sufficient, or should we specifically ensure that `block.blockTime` is defined, too?

Reply:
Great catch... I _assume_ this is because validators aren't required to submit timestamps for each slot, so there might be cases where none submit their timestamp.

Updated: https://github.com/across-protocol/sdk/pull/1115/commits/6616695371c9e212299cf34be4dfb631f22c2be6